torch/testing/_internal/common_nn.py

```diff
@@ -5033,18 +5028,36 @@ def __init__(self, *args, **kwargs):
         self.with_tf32 = kwargs.get('with_tf32', False)
         self.tf32_precision = kwargs.get('tf32_precision', 0.001)
         self.test_cpu = kwargs.get('test_cpu', True)
+        self.has_sparse_gradients = kwargs.get('has_sparse_gradients', False)
+
+    def _check_gradients(self, test_case, module, input_tuple):
+        params = tuple(x for x in module.parameters())
+        num_inputs = len(input_tuple)
+
+        def fn_to_gradcheck(*inputs_and_params, **kwargs):
+            assert not kwargs
+            return test_case._forward(module, inputs_and_params[:num_inputs])
+
+        # gradcheck doesn't support operators that take in dense inputs but
+        # return sparse parameters. This only happens in the case of nn.Embedding
+        # and nn.EmbeddingBag. Instead, we call `self.check_jacobian`, which
+        # is a slightly different version of gradcheck that can handle this.
+        if self.has_sparse_gradients:
+            assert len(input_tuple) == 1
+            test_input_jacobian = torch.is_floating_point(input_tuple[0])
```

Comment:
This is a 1:1 correspondence with the flag that used to exit in each definition?

Reply:
Yes, all of the tests that used to have `test_input_jacobian=True` had non-floating-point inputs.